### ISSUE TITLE
Trying to Push as the wrong user.

### ISSUE THREAD

**Issue by modusinternet** [atom/github#2395]
A couple of months ago I worked on a website that was repoed on a private GitHub account.  I was granted access to work on it as a contributor and successfully used Atom on Windows to connect to it for about two weeks.  When my work was done I was removed as a contributor from the repo which I then removed entirely from my copy of Atom, my list of repos in GitHub Desktop and my harddrive.  (I thought)

Since then I can not use Atom to push updates to any of my other projects without using GitHub Desktop for the last step.  Meaning, I'm used to just Saving my changes, Staging, Committing and Pushing them all within Atom, but not anymore.  Now I get this error every time I try to Push and I don't know how to fix it...

![70838949-6a717180-1dbf-11ea-94d1-34586ba9c27a](https://user-images.githubusercontent.com/4643367/72839123-83830300-3c46-11ea-9452-a4cf7feb7efa.png)

'boatwright' is the name of the repo that I WAS working on months ago and I'm not trying to push my updates to them.  I'm trying to push to my own repo 'modusinternet/modus_dev'.  Does anyone know where in Atom is this information stored?

Regards,

Vince

**Comment by smashwilson** [comment]
If you're able to push from GitHub Desktop, it's likely that you're using the token acquired from the GitHub tab to push. Try running `GitHub: Logout` from the command palette and see if that straightens things out.

**Comment by modusinternet** [comment]
Nothing happens when I do that and nothing changes in my ability to push updates back to my repos.  I still have to do the final push using GitHub Desktop.

**Comment by smashwilson** [comment]
Okay, then it must be using credentials cached by git itself in your credential manager. The specific of clearing credentials stored there vary depending on your system configuration -- I think you can usually open your OS keychain and delete them from there?

I think you can also open up a command prompt and type:

```
git credential-manager delete
```

If _that_ doesn't work, I'll need more information to help you out. If you can paste the result from this command, that would help:

```
git config --get-all credential.helper
```

**Comment by modusinternet** [comment]
I'm on Windows 10, running Atom v1.43.0 x64, and GitHub Desktop v2.2.4.

I have no idea what a 'credential manager' or an 'OS keychain' is and running 'git credential-manager delete' from a Windows PowerShell, Git Bash, Git CMD, or Git Shell all return 'fatal: Unable to parse target URI'.

Running 'git config --get-all credential.helper' in Git Shell returns '!github --credentials'.

**Comment by modusinternet** [comment]
And looking for 'boatwright' in the Windows Registry turns up nothing either.

**Comment by smashwilson** [comment]
Okay, I'm on my Windows machine now, which makes this a _little_ easier to debug. Try this:

1. Open "Credential Manager" from your start menu:

![credential manager](https://user-images.githubusercontent.com/17565/72846623-f0aa8e80-3c6e-11ea-932b-ed2016818d49.png)

2. Choose "Windows Credentials" in the boxes at the top:

![windows credentials](https://user-images.githubusercontent.com/17565/72846665-0a4bd600-3c6f-11ea-8254-7d16cb9d66f1.png)

3. Look under "Generic Credentials" for keys starting with "atom-github-git-meta" or "atom-github-git":

![credential keys](https://user-images.githubusercontent.com/17565/72846747-2e0f1c00-3c6f-11ea-8578-76550e813078.png)

4. Expand each one and click "remove":

![click remove](https://user-images.githubusercontent.com/17565/72846807-4ed77180-3c6f-11ea-980d-aa73fe3b61ae.png)

**Comment by modusinternet** [comment]
You da man smashwilson!

I found it, not exactly saved as you describe but you got me there.


![Untitled](https://user-images.githubusercontent.com/4643367/72848157-d8c81000-3c58-11ea-900a-55fc97fbf86d.png)


Once I delete this entry I tried pushing again in Atom and got this popup.


![Untitled2](https://user-images.githubusercontent.com/4643367/72848210-f6957500-3c58-11ea-8700-26c5081bfe34.png)


I canceled it just to see what would happen and this next box popped up.


![Untitled3](https://user-images.githubusercontent.com/4643367/72848247-0ca33580-3c59-11ea-8a12-f3b467155653.png)


I went ahead and filled that in and it worked perfectly.

Incidentally, here is what I see at the top of my Windows Credentials now.


![Untitled4](https://user-images.githubusercontent.com/4643367/72848368-45dba580-3c59-11ea-8a4a-8827e4972c44.png)


So it looks like skipping the first login has added a new entry called 'username', not sure if that will be an issue down the road but at least I know where to go now if I need to nuke it.

Thank you so much for your help, this had been bothering me for almost two months now.

Sincerely,

Vince
